### PR TITLE
configure_openssl_cryptopolicy: align remediations with rule description

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- change the ".include"  line which is used in remediations to be coherent with rule descriptions for different products

#### Rationale:

- remediations were always using the include statement without the "=" sign, whereas rule description used different statements for different products. This PR aligns remediations with the description.

#### Review Hints:

1. ./build_product rhel8 rhel9
2. compare Bash and Ansible remediations for configure_openssl_crypto_policy between rhel8 and rhel9 products
3. compare rule descriptions of configure_openssl_crypto_policy between rhel8 and rhel9 product
4. descriptions should match remediations in respective products

### CODE REVIEW

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:12]: I think it would be good for the simplicity and prosperity of this rule to simplify the remediation without using `=` in all products. There is no technical impact on it.

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/bash/shared.sh
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/bash/shared.sh:5]: For these products, if the `/etc/pki/tls/openssl.cnf` file was previously configured and is technically compliant but using the `=` signal, like this:
`.include = /etc/crypto-policies/back-ends/opensslcnf.config`

The conditional on line 26 will lead to a duplicated configuration, one using `=` and other not using `=`.
Can this cause any trouble or is one line simply ignored? Besides that, would it be better to check both options and only change if none is there or even ensure only the product preference.

Something similar will happen with the Ansible remediation.

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
There is a concern about duplicating the configuration during the remediation.

**Comment by marcusburghardt** [comment]
In a private discussion we saw benefits and drawbacks on aligning the syntax for all products, so the remediation would always ensure a line without the `=` signal,
like:
`.include /etc/crypto-policies/back-ends/opensslcnf.config`

instead of:
`.include = /etc/crypto-policies/back-ends/opensslcnf.config`

Both syntax are valid while the longer option (with `=`) is acceptable by compatibility purposes only. So, there is no technical impact on choosing any of them. The OVAL already accept both, but the remediation may differ based on product. I suggested to simplify and keep the same syntax everywhere. The drawbacks would be a misalignment with some benchmarks which explicitly ask a syntax. In short, if this is the case, the respective benchmarks should update their requirement descriptions to be indifferent to this matter, even if using one specific syntax as example.

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/bash/shared.sh
Comment (vojtapolasek) [reply linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/bash/shared.sh:5]: @marcusburghardt  Thanks for bringing this up.I have modified both Bash and Ansible remediation in a way so that they respect both options (with our without =) in the same way the OVAL check does. Only in case the option is malformed in some other way or missing they remediate. Do you agree with this? Another way would be to modify OVAL instead and use different regex for RHEL 9, OL 9 and Fedora. Both options seem fine to me.

Comment (marcusburghardt) [reply linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/bash/shared.sh:5]: The proposed fix is fine. Thanks @vojtapolasek .

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:22]:
```suggestion
- name: "{{{ rule_title }}} - Search for crypto_policy Section"
```

Comment (marcusburghardt) [reply linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:22]: Tasks names should be in Title Case: https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#ansible

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:29]:
```suggestion
- name: "{{{ rule_title }}} - Test for Crypto Policy Section Together With Include Directive"
```

Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:29]:
```suggestion
- name: "{{{ rule_title }}} - Search for crypto_policy Section Together With .include Directive"
```

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:36]:
```suggestion
- name: "{{ rule_title }} - Add .include Line for opensslcnf.config File in crypto_policy Section"
```

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml
Comment (marcusburghardt) [inline linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:46]:
```suggestion
- name: "{{{ rule_title }}} - Add crypto_policy Section With .include for opensslcnf.config File"
```

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
I have only some minor suggestions for the Ansible tasks names. They should follow the Title Case in alignment to our Style Guides.

File: linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml
Comment (vojtapolasek) [reply linux_os/guide/system/software/integrity/crypto/configure_openssl_crypto_policy/ansible/shared.yml:22]: I have accepted your suggestions.

**Review by marcusburghardt — APPROVED** [review]
Thanks @vojtapolasek

**Comment by marcusburghardt** [comment]
The CI tests were broken after the #10842 was merged and this PR is blocked by this.

In this particular PR, I have successfully tested it in local VMs.
Before the #10842 PR , CI tests also passed and after the #10842 PR, commits were only about Ansible tasks names, without technical impact in the remediation. Therefore, I am sure there won't be negative consequences to waive the required tests. It can be safely merged as it is, if necessary.

**Pull request merged**
